### PR TITLE
[Merged by Bors] - Add a MNIST sample to demonstrate how to use Adlik

### PR DESCRIPTION
**Pull request by shiyingjin**

### CODE REVIEW

File: samples/simpleMNIST/README.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+3. Install model_compiler: 
+
+   ```
+   cd {Adlik_root_dir}/model_compiler
+   pip3 install .
+   ```
+
+# Running the Sample
+1. Train the model and save:
```
Comment (amadeus-zte) [inline samples/simpleMNIST/README.md]: shoule be "save checkpoints file" or "save .h5 file"

File: model_compiler/src/model_compiler/config.py [hunk 2]
```diff
@@ -16,8 +16,8 @@ class _Config:
     def __init__(self, **kwargs):
         if 'job_id' not in kwargs:
             raise Exception('Not found job id in config, config: %s' % kwargs)
-        if 'callback' not in kwargs:
-            raise Exception('Not found callback in config, config: %s' % kwargs)
+        # if 'callback' not in kwargs:
+        #     raise Exception('Not found callback in config, config: %s' % kwargs)
```
Comment (EFanZh) [inline model_compiler/src/model_compiler/config.py]: Remove these two lines if not used.

File: samples/simpleMNIST/train_model.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+def _train():
+    x_train, y_train, x_test, y_test = process_dataset()
+    model = create_model()
+    model.fit(x_train, y_train, epochs=10, verbose=1)
+    model.evaluate(x_test, y_test)
+    save_path = os.path.join(os.path.dirname(__file__), 'model', 'mnist.h5')
+    dir_name = os.path.dirname(save_path)
+    if not os.path.exists(dir_name):
+        os.makedirs(dir_name)
```
Comment (EFanZh) [inline samples/simpleMNIST/train_model.py]: Use `exist_ok=True` argument.

File: samples/simpleMNIST/mnist_client.py [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+    return input_.name, output.name, c, h, w, input_.format, tensor_dtype_to_np_dtype(input_.data_type)
+
+
+def _gen_input_data(data_format, dtype, c, h, w):
+    if os.path.isdir(FLAGS.image_filename):
+        filenames = [os.path.join(FLAGS.image_filename, f)
+                     for f in os.listdir(FLAGS.image_filename)
+                     if os.path.isfile(os.path.join(FLAGS.image_filename, f))]
+    else:
+        filenames = [FLAGS.image_filename, ]
```
Comment (EFanZh) [inline samples/simpleMNIST/mnist_client.py]: Remove the comma.

File: samples/simpleMNIST/README.md [hunk 5]
```diff
@@ -0,0 +1,87 @@
+# About this sample
+
+This sample demonstrates how to use Adlik to compile serving model and deploy serving service end to end.
+
+## Installing Prerequisites
```
Comment (EFanZh) [inline samples/simpleMNIST/README.md]: Use lower case “prerequisites”.

File: samples/simpleMNIST/README.md [hunk 6]
```diff
@@ -0,0 +1,87 @@
...
+2. Install clients pip package:
+
+   ```sh
+   pip3 install {directory-of-pip-package}/adlik_serving_api-0.0.0-py2.py3-none-any.whl
+   ```
+
+3. Install model_compiler:
+
+   ```sh
+   cd {Adlik_root_dir}/model_compiler
```
Comment (EFanZh) [inline samples/simpleMNIST/README.md]: Either use dash (“-”) character or underscore character (“_”).

File: samples/simpleMNIST/README.md [hunk 7]
```diff
@@ -0,0 +1,87 @@
...
+   pip3 install .
+   ```
+
+## Running the Sample
```
Comment (EFanZh) [inline samples/simpleMNIST/README.md]: Use lower case “sample”.

**Pull request closed without merging**